bin/find-device-port-macos

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env perl
+
+use warnings;
+use strict;
+
+# ioreg might be more machine-readable than system_profiler, but I haven't been able to
+# get it to produce useful output
+my @output = qx(/usr/sbin/system_profiler SPUSBDataType);
+
+my $parse_state = 0;
+my $device      = {};
+
+LINE: foreach my $line (@output) {
+
+    chomp $line;
+
+    if ( $parse_state == 0 ) {
+        if ( $line =~ m/Model 01/ ) {
+            $parse_state = 1;
+            next LINE;
+        }
+    }
+
+    if ( $parse_state == 1 ) {
+        if ( $line =~ m/^\s*$/ ) {
+            $parse_state = 2;
+            next LINE;
+        }
+    }
+
+    if ( $parse_state == 2 ) {
+        if ( $line =~ m/Serial Number: (.+)$/ ) {
+            $device->{'serial_number'} = $1;
+            next LINE;
+        }
+        if ( $line =~ m/Location ID: (.+)$/ ) {
+            $device->{'location_id'} = $1;
+            next LINE;
+        }
+        if ( $line =~ m/Product ID: (.+)$/ ) {
+            $device->{'product_id'} = $1;
+            next LINE;
+        }
+        if ( $line =~ m/Vendor ID: (.+)$/ ) {
+            $device->{'vendor_id'} = $1;
+            next LINE;
+        }
+        if ( $line =~ m/^\s*$/ ) {
+            last LINE;
+        }
+    }
+
+}
+
+die "Can't find Model 01" if ( $device == {} );
+
+my $serial_port_name = "";
+
+if ( exists( $device->{'serial_number'} ) ) {
+    $serial_port_name = "/dev/cu.usbmodem" . $device->{'serial_number'};
```

Comment:
I think this section covers all the cases in the previous `kaleidoscope-builder.conf` except the last one. We could also test `/dev/tty.usbmodem*`, but I don't know if any of our devices could show up with that name, and the previous version of `kaleidoscope-builder.conf` didn't use those names.